applications/pelion_client/README.rst

```diff
@@ -0,0 +1,516 @@
+For more information about the provisioning process, see `Provisioning devices for Pelion Device Management`_ in the Pelion documentation.
+
+The application follows the developer flow provisioning process.
+This process involves downloading the source file that contains a valid certificate.
+The file is available from Pelion's Device Management Portal and is used at build time to embed the required information to the application binary.
+
+During the first boot, or whenever the Pelion storage partition is erased (:c:func:`fcc_storage_delete`), the storage will be populated with the credentials passed at build time.
+This is done by calling :c:func:`fcc_developer_flow` right after the Factory Configurator Client (FCC) is initialized (:c:func:`fcc_init`).
+For more information about these functions, see the `Pelion's factory_configurator_client.h`_ file reference page.
+
+Using the valid certificate will allow the device to successfully register (connect) to the Pelion cloud.
+At this stage, the device will be assigned a new device identity, under which it can be recognized later on.
+
+Run time flow overview
+======================
+
+The following sections describe the Pelion client application operational flow at run time, after provisioning.
+
+.. contents::
+   :local:
+   :depth: 4
+
+Factory Configuration Client initialization
+-------------------------------------------
+
+After the board boot sequence, the application's ``pelion_fcc`` module initializes Pelion's FCC.
+At this stage, the developer flow is enabled and the certificate is loaded to the Pelion storage partition (if needed).
+
+Network backend initialization
+------------------------------
+
+In parallel to the FCC initialization, the application initializes the network backend.
+The appropriate net module connects the required callbacks and starts either OpenThread (``net_ot``, using Zephyr's :ref:`zephyr:thread_protocol_interface`) or the link control (``net_lte``, using the :ref:`lte_lc_readme`).
+
+The network module tracks the connection state and informs the rest of the application if the connection is established or the network is disconnected.
+
+Pelion module initialization
+----------------------------
+
+Pelion's client object is initialized from the Pelion module when the FCC is initialized.
+The module attaches several client callbacks that are required for informing the application about the connection status with the Pelion cloud.
+
+After initialization, the application starts the process of creating resource objects.
+The application broadcasts the event that triggers the resource creation and all other modules wishing to create resources must act at this time.
+
+OMA resource creation
+---------------------
+
+After the cloud client is initialized by the application, it sends ``create_object_event``, which starts the creation of objects with their related OMA resources.
```

Comment:
@pdunaj , would "Pelion Device Management Client" be a good name for the application? After all, the application implements it, so it would make sense to have this more or less consistent on both sides. Alternative proposal (I like it more, but it's longer): "Pelion Device Management Client for the nRF Connect SDK"...